Missing JavaDoc for some public classes and methods

> Property (SE): the program is properly documented
> 
> Assessment: The grader randomly opens application files (not test files) to check that all public classes and methods are well documented ([hints](http://www.monperrus.net/martin/how-to-write-good-API-documentation)). In addition, the grader checks that the corresponding API documentation has been generated in a browsable format (eg HTML when using Javadoc).

`gradle javadoc` is currently able to generate documentation at `./app/build/docs/javadoc/index.html`, but some public classes and methods are missing documentation. I believe the only public class with JavaDoc so far is `PipelineUpdateRequest`

excerpts from `$ gradle javadoc`
```
App.java:6: warning: no comment
public class App {
       ^
App.java:6: warning: use of default constructor, which does not provide a comment
public class App {
       ^
App.java:7: warning: no comment
    public String getGreeting() {
                  ^
App.java:11: warning: no comment
    public static void main(String[] args) {
                       ^
ContinuousIntegrationServer.java:18: warning: no comment
public class ContinuousIntegrationServer extends AbstractHandler {
       ^
ContinuousIntegrationServer.java:18: warning: use of default constructor, which does not provide a comment
public class ContinuousIntegrationServer extends AbstractHandler {
       ^
ContinuousIntegrationServer.java:85: warning: no comment
    public static void startServer() throws Exception {
                       ^
PipelineUpdateRequest.java:102: warning: no @throws for java.lang.InterruptedException
    public int send() throws InterruptedException, IOException {
               ^
PipelineUpdateRequest.java:102: warning: no @throws for java.io.IOException
    public int send() throws InterruptedException, IOException {
               ^
PushEvent.java:29: warning: no comment
public class PushEvent {
       ^
PushEvent.java:29: warning: use of default constructor, which does not provide a comment
public class PushEvent {
       ^
11 warnings
```

--- a/app/src/main/java/ciserver/App.java
+++ b/app/src/main/java/ciserver/App.java
@@ -3,11 +3,24 @@
  */
 package ciserver;
 
+/**
+ * This class is the initial class used to start the whole CI server program
+ */
 public class App {
+    /**
+     * Gets a greeting, in this case "Hello world!"
+     *
+     * @return String of the greeting to print
+     */
     public String getGreeting() {
         return "Hello World!";
     }
 
+    /**
+     * Print a greeting then launch the CI server
+     *
+     * @param args the arguments to the program, not used
+     */
     public static void main(String[] args) {
         System.out.println(new App().getGreeting());
 

--- a/app/src/main/java/ciserver/App.java
+++ b/app/src/main/java/ciserver/App.java
@@ -19,7 +19,7 @@ public class App {
     /**
      * Print a greeting then launch the CI server
      *
-     * @param args the arguments to the program, not used
+     * @param args the command line arguments to the program; not used
      */
     public static void main(String[] args) {
         System.out.println(new App().getGreeting());

--- a/app/src/main/java/ciserver/ContinuousIntegrationServer.java
+++ b/app/src/main/java/ciserver/ContinuousIntegrationServer.java
@@ -56,7 +56,7 @@ public class ContinuousIntegrationServer extends AbstractHandler {
     }
 
     /**
-     * See
+     * Handle an HTTP request to the CI server. See
      * <a
      * href=https://www.eclipse.org/jetty/javadoc/jetty-9/org/eclipse/jetty/server/handler/AbstractHandler.html#handle(java.lang.String,org.eclipse.jetty.server.Request,javax.servlet.http.HttpServletRequest,javax.servlet.http.HttpServletResponse)>documentation
      * of the parent interface</a>

--- a/app/src/main/java/ciserver/ContinuousIntegrationServer.java
+++ b/app/src/main/java/ciserver/ContinuousIntegrationServer.java
@@ -56,6 +56,10 @@ public class ContinuousIntegrationServer extends AbstractHandler {
     }
 
     /**
+     * See
+     * <a
+     * href=https://www.eclipse.org/jetty/javadoc/jetty-9/org/eclipse/jetty/server/handler/AbstractHandler.html#handle(java.lang.String,org.eclipse.jetty.server.Request,javax.servlet.http.HttpServletRequest,javax.servlet.http.HttpServletResponse)>documentation
+     * of the parent interface</a>
      * {@inheritDoc}
      */
     public void handle(String target,

--- a/app/src/main/java/ciserver/ContinuousIntegrationServer.java
+++ b/app/src/main/java/ciserver/ContinuousIntegrationServer.java
@@ -16,6 +16,11 @@ import org.eclipse.jetty.server.handler.AbstractHandler;
 import com.google.gson.Gson;
 import com.google.gson.JsonSyntaxException;
 
+/**
+ * This class is the core class for running the CI server.
+ * The CI server compiles and tests Java code with Gradle, and can set commit
+ * statuses in GitHub
+ */
 public class ContinuousIntegrationServer extends AbstractHandler {
 
     private final Gson gson = new Gson();
@@ -50,6 +55,9 @@ public class ContinuousIntegrationServer extends AbstractHandler {
         System.out.println("Commit status update sent successfully.");
     }
 
+    /**
+     * {@inheritDoc}
+     */
     public void handle(String target,
             Request baseRequest,
             HttpServletRequest request,
@@ -84,7 +92,11 @@ public class ContinuousIntegrationServer extends AbstractHandler {
         response.getWriter().println("CI job done");
     }
 
-    // Used to start the CI server in command line
+    /**
+     * Used to start the CI server in command line
+     * 
+     * @throws Exception if the server cannot be started
+     */
     public static void startServer() throws Exception {
         Server server = new Server(8080);
         server.setHandler(new ContinuousIntegrationServer());

--- a/app/src/main/java/ciserver/PipelineUpdateRequest.java
+++ b/app/src/main/java/ciserver/PipelineUpdateRequest.java
@@ -132,6 +132,13 @@ public class PipelineUpdateRequest {
      * containing information about the success status of a certain build.
      *
      * @return the HTTP response formatted as a String.
+     *
+     * @throws InterruptedException {@link java.net.http.HttpClient#send(HttpRequest, HttpResponse.BodyHandler)
+     *                              if the operation is interrupted}
+     * @throws IOException          {@link java.net.http.HttpClient#send(HttpRequest, HttpResponse.BodyHandler)
+     *                              if an I/O error occurs when sending or
+     *                              receiving}
+     *
      **/
     public HttpResponse<String> send() throws InterruptedException, IOException {
         Gson gson = new Gson();

--- a/app/src/main/java/ciserver/PushEvent.java
+++ b/app/src/main/java/ciserver/PushEvent.java
@@ -5,41 +5,41 @@ import java.util.ArrayList;
 import com.google.gson.annotations.SerializedName;
 
 class Repository {
-	String name;
-	String full_name;
+    String name;
+    String full_name;
 
-	@SerializedName("clone_url")
-	String cloneUrl;
+    @SerializedName("clone_url")
+    String cloneUrl;
 }
 
 class Author {
-	String name;
-	String email;
-	String username;
+    String name;
+    String email;
+    String username;
 }
 
 class Commit {
-	String id;
-	String url;
+    String id;
+    String url;
 
-	Author author;
+    Author author;
 
-	ArrayList<String> added; // added files
-	ArrayList<String> removed; // removed files
-	ArrayList<String> modified; // modified files
+    ArrayList<String> added; // added files
+    ArrayList<String> removed; // removed files
+    ArrayList<String> modified; // modified files
 }
 
 /**
  * This class represents the relevant information from a Push Event to GitHub
  */
 public class PushEvent {
-	String ref; // hold branch name
-	String compare; // link to see changes
+    String ref; // hold branch name
+    String compare; // link to see changes
 
-	Author pusher;
-	Repository repository;
-	ArrayList<Commit> commits;
+    Author pusher;
+    Repository repository;
+    ArrayList<Commit> commits;
 
-	@SerializedName("head_commit")
-	Commit headCommit;
+    @SerializedName("head_commit")
+    Commit headCommit;
 }

--- a/app/src/main/java/ciserver/PushEvent.java
+++ b/app/src/main/java/ciserver/PushEvent.java
@@ -32,10 +32,13 @@ class Commit {
 /**
  * This class represents the relevant information from a Push Event to GitHub.
  *
- * All fields are set up to match the "push" event http request JSON data from
- * GitHub so that it can be parsed with the gson library. See <a
+ * This class is used by Gson to parse the JSON payload of the GitHub push
+ * webhook.
+ * Gson will parse the parts of the JSON payload that match the fields of this
+ * class. See <a
  * href=https://docs.github.com/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#push>the
- * documentation on GitHub</a>
+ * documentation on GitHub</a> for information about the class fields/JSON
+ * payload.
  */
 public class PushEvent {
     String ref; // hold branch name

--- a/app/src/main/java/ciserver/PushEvent.java
+++ b/app/src/main/java/ciserver/PushEvent.java
@@ -30,7 +30,12 @@ class Commit {
 }
 
 /**
- * This class represents the relevant information from a Push Event to GitHub
+ * This class represents the relevant information from a Push Event to GitHub.
+ *
+ * All fields are set up to match the "push" event http request JSON data from
+ * GitHub so that it can be parsed with the gson library. See <a
+ * href=https://docs.github.com/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#push>the
+ * documentation on GitHub</a>
  */
 public class PushEvent {
     String ref; // hold branch name

--- a/app/src/main/java/ciserver/PushEvent.java
+++ b/app/src/main/java/ciserver/PushEvent.java
@@ -29,6 +29,9 @@ class Commit {
 	ArrayList<String> modified; // modified files
 }
 
+/**
+ * This class represents the relevant information from a Push Event to GitHub
+ */
 public class PushEvent {
 	String ref; // hold branch name
 	String compare; // link to see changes

